lib/routes/sohu/mobile.ts

```diff
@@ -0,0 +1,109 @@
+import { Route } from '@/types';
+import ofetch from '@/utils/ofetch';
+import cache from '@/utils/cache';
+import * as cheerio from 'cheerio';
+import logger from '@/utils/logger';
+
+export const route: Route = {
+    path: '/mobile',
+    categories: ['new-media', 'popular'],
+    example: '/sohu/mobile',
+    parameters: {},
+    features: {
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    radar: [
+        {
+            source: ['m.sohu.com/limit'],
+            target: '/mobile',
+        },
+    ],
+    name: '首页新闻',
+    maintainers: ['asqwe1'],
+    handler,
+    description: '订阅手机搜狐网的首页新闻',
+};
+
+async function handler() {
+    try {
+        const response = await ofetch('https://m.sohu.com/limit', {
+            headers: {
+                'User-Agent': 'Mozilla/5.0 (iPhone; CPU iPhone OS 15_0 like Mac OS X)'
+            }
+        });
+        // 从HTML中提取JSON数据
+        const $ = cheerio.load(response);
+        const jsonScript = $('script:contains("WapHomeRenderData")').html();
+        const jsonMatch = jsonScript?.match(/window\.WapHomeRenderData\s*=\s*({.*})/s);
+        if (!jsonMatch?.[1]) {
+            throw new Error('WapHomeRenderData 数据未找到');
+        }
+        const renderData = JSON.parse(jsonMatch[1]);
+        const list = extractPlateBlockNewsLists(renderData)
+            .map((item) => ({
+                title: item.title,
+                link: new URL(item.link || item.url, 'https://m.sohu.com').href,
+            }))
+            .filter((item) => item?.title && item?.link);
+        const items = await Promise.all(
+            list.map((item) =>
+                cache.tryGet(item.link, async () => {
+                    try {
+                        const detailResp = await ofetch(item.link);
+                        const $d = cheerio.load(detailResp);
+                        return {
+                            ...item,
+                            description: $d('#articleContent').first().html(),
+                            pubDate: parseDate($d('.time').text()) // 添加时间解析
+                        };
```

Comment:
Consider handling video post like `https://m.sohu.com/a/890619030_100145529`, gallery post like `https://m.sohu.com/a/890097358_598290` and short post like `https://m.sohu.com/xtopic/TURBeE1qY3hOVE01`